### PR TITLE
Copter: Solo buttons

### PR DESCRIPTION
**Pull request by squilter**
The Solo controller has some buttons on the front to take off, land and pause. I transferred that code over from the ardupilot-solo repository. I didn't transfer the functionality of the pause button, because I don't think pause mode exists in master.

### CODE REVIEW

**Comment by magicrub** [comment]
@squilter No description and commit msgs are blank. Can you fill-in the blank(s) please?

**Comment by squilter** [comment]
The Solo controller has some buttons on the front to take off, land and pause.  I transferred that code over from the ardupilot-solo repository.  I didn't transfer the functionality of the pause button, because I don't think pause mode exists in master.

File: ArduCopter/APM_Config.h [hunk 1]
```diff
@@ -39,6 +39,7 @@
 // features below are disabled by default on all boards
 //#define SPRAYER               ENABLED             // enable the crop sprayer feature (two ESC controlled pumps the speed of which depends upon the vehicle's horizontal velocity)
 //#define PRECISION_LANDING     ENABLED             // enable precision landing using companion computer or IRLock sensor
+//#define SOLO_BUTTONS          ENABLED
```
Comment (squilter) [inline ArduCopter/APM_Config.h]: Maybe there should just an ifdef called `SOLO_FEATURES`.  There will probably be many more solo things that we'll want to hide behind ifdefs.

**Comment by R-Lefebvre** [comment]
Pause mode might exist in Master. What exactly is it?  We have Stop mode, I created that for Solo, not sure what the final integration was like.

**Comment by squilter** [comment]
Oh yeah, I meant Stop mode.  I only see `control_stop` in ardupilot-solo, I don't see it in master.

**Comment by R-Lefebvre** [comment]
It's called Control_Brake on Master.

**Comment by rmackay9** [comment]
@squilter, the requirement for copter. to be prefixed came with the move to .cpp files.  We moved everything from the global scope into the Copter object.

**Comment by OXINARF** [comment]
The result is always set to accept even when mode can't be set. Is that the intended behavior?

**Comment by rmackay9** [comment]
I think the GCS part of this PR has gone in now: https://github.com/diydrones/ardupilot/pull/3398
There's still the 2nd commit though that's not in.

**Comment by jschall** [comment]
@rmackay9 yep - I'll almost certainly review this PR or make a new one next week.

**Comment by jschall** [comment]
I made a new one: https://github.com/diydrones/ardupilot/pull/3426

**Pull request closed without merging**